src/com/owncloud/android/ui/activity/UploadFilesActivity.java

```diff
@@ -171,9 +189,46 @@ public void onCreate(Bundle savedInstanceState) {
     public static void startUploadActivityForResult(Activity activity, Account account, int requestCode) {
         Intent action = new Intent(activity, UploadFilesActivity.class);
         action.putExtra(EXTRA_ACCOUNT, (account));
+        action.putExtra(REQUEST_CODE_KEY,requestCode);
         activity.startActivityForResult(action, requestCode);
     }
 
+    private void uploadFromCamera(){
+        Intent pictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (pictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(pictureIntent, REQUEST_IMAGE_CAPTURE);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode,int resultCode,Intent capturedData){
+        if(requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK){
+            Bundle capturedImageData = capturedData.getExtras();
+            Bitmap capturedImage = (Bitmap) capturedImageData.get("data");
+            Uri capturedImageUri = getImageUri(getApplicationContext(),capturedImage);
+            capturedImagePath = getRealPathFromUri(capturedImageUri);
```

Comment:
Instead of declaring capturedImagePath as global, you can declare and initialize it here and pass it to CheckAvailableSpaceTask(), look the answer of [https://stackoverflow.com/questions/3075009/android-how-can-i-pass-parameters-to-asynctasks-onpreexecute/3077050](url)